packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -44,7 +44,26 @@ class PlatformViewsRegistry {
   ///
   /// Typically a platform view identifier is passed to a platform view widget
   /// which creates the platform view and manages its lifecycle.
-  int getNextPlatformViewId() => _nextPlatformViewId++;
+  int getNextPlatformViewId() {
+    // On the Android side, the interface exposed to users uses 32-bit integers.
+    // See https://github.com/flutter/engine/pull/39476 for more details.
+
+    // We can safely assume that a Flutter application will not require more
+    // than MAX_INT32 platform views on the fly throughout its lifetime.
```

Comment:
This comment is inconsistent with the next comment and the code; if it's safe to assume that we won't need more than MAX_INT32 platform views over the course of the application's lifetime, then we don't code to do rollover because it'll never be run; we can simply assert.

If what you actually mean is simultaneous views, and you actually want to handle that, this code isn't sufficient; you would actually need to track which views IDs are still active, because just because there's been rollover doesn't mean that *every* previous ID is gone. If someone cycling through more than a billion platform views is plausible, then it's equally plausible that they do so while still using, say, view number 1.

Reply:
Yes, you're right. We can simply use an assertion. :)